nucypher/config/storages.py

```diff
@@ -519,17 +535,18 @@ def initialize(self) -> bool:
             except FileNotFoundError:
                 raise self.NodeStorageError("There is no existing configuration at {}".format(self.root_dir))
 
-        return bool(all(map(os.path.isdir, (self.root_dir, self.metadata_dir, self.certificates_dir))))
-
 
 class TemporaryFileBasedNodeStorage(LocalFileBasedNodeStorage):
     _name = 'tmp'
 
     def __init__(self, *args, **kwargs):
         self.__temp_metadata_dir = None
         self.__temp_certificates_dir = None
+        self.__temp_root_dir = None
+        self.initialize()
```

Comment:
how come `self.iniitalize()` is called in here?

Reply:
Because we need to create the directory structure prior calling the parent's constructor.

Reply:
Is the reason then for the separate initialize() function, and not just doing it in the constructor, that it allows some sort of re-initialize functionality...i.e. `initialize()` called more than once recreate the temp directory etc...?

Reply:
I think so, but the method exists from the original construction, back in 2018. I simply respected the API

Reply:
Of course we can discuss later whether the original strategy in this class is the best way to do what it is trying to do - but for the purposes of this PR we don't have to discuss that right now 😅

Reply:
I think you're totally right, thanks for noticing!
✅